File: Interfaces/IAuthService.cs
using cms_webapi.DTOs;
using cms_webapi.Models;

namespace cms_webapi.Interfaces
{
    public interface IAuthService
    {
        Task<ApiResponse<User>> GetUserByUsernameAsync(string username);
        Task<ApiResponse<User>> GetUserByIdAsync(int id);
        Task<ApiResponse<User>> RegisterUserAsync(RegisterDto registerDto);
        Task<ApiResponse<string>> LoginAsync(LoginDto loginDto);
    }
}

File: WebApi/Interfaces/IAuthService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { LoginDto, LoginResponseDto } from '../Models/AuthDto';
import { CreateUserDto, UserDto } from '../Models/UserDto';
import { User } from '../Models/User';

export interface IAuthService {
  login(loginDto: LoginDto): Promise<ApiResponse<LoginResponseDto>>;
  register(createUserDto: CreateUserDto): Promise<ApiResponse<UserDto>>;
  validateUser(usernameOrEmail: string, password: string): Promise<User | null>;
  emailExists(email: string): Promise<boolean>;
  usernameExists(username: string): Promise<boolean>;
}